app/models/user.rb

```diff
@@ -0,0 +1,16 @@
+class User < ActiveRecord::Base
+  include EmailNormalization
+
+  belongs_to :manager, class_name: :User
+  has_many :employees, class_name: :User, foreign_key: :manager_id
+
+  validates :email, presence: true, format: /\A.+@.+\z/, uniqueness: true
```

Comment:
@jasiek Minimal viable validation. Check that the email goes in the right field. Everything else is waste, as email address parsing is complex, and the only way to be sure is to send an email.